### PR TITLE
cpu/stm32f[0|1|3] unify stmclk with stm32_common

### PR DESCRIPTION
**Pull request by vincent-d**
Based on #7477 

Another step in unifying clock config for stm32.
F0 and F1 are quite similar for clock config.

I can test only on nucleo-f091 and nucleo32-f042

I don't plan to work on stm32lx cpu

### CODE REVIEW

**Comment by aabadie** [comment]
Ran a few tests. examples/default, tests/xtimer_msg, tests/driver_sx127x:
- nucleo-f070/nucleo-f072: build fails because there's no LSE I think, the board periph_conf needs to be updated, see https://github.com/OTAkeys/RIOT/pull/8
- nucleo-f103: works

**Comment by aabadie** [comment]
doesn't build with nucleo32-f031

**Comment by vincent-d** [comment]
@aabadie I updated only nucleo-f091 and nucleo-f103 so far, others need to be ported too

**Comment by vincent-d** [comment]
Indeed, nucleo32 do not have HSE IIRC

**Comment by vincent-d** [comment]
updated nucleo32-f031 / f042

**Comment by aabadie** [comment]
great! unfortunately I won't be able to test before next week...

**Comment by vincent-d** [comment]
It turned out f3 is very similar to f0 and f1, so merging it as well.

**Comment by vincent-d** [comment]
Rebased on master

File: cpu/stm32f1/stmclk.c [hunk 1]
```diff
@@ -27,7 +27,7 @@
 #include "cpu.h"
```
Comment (haukepetersen) [inline cpu/stm32f1/stmclk.c]: can't we remove this file completely now?

Comment (vincent-d) [reply cpu/stm32f1/stmclk.c]: I forgot it, but we must remove it, for sure!

File: cpu/stm32_common/doc.txt [hunk 2]
```diff
@@ -8,10 +8,11 @@
  * @ref cpu_stm32f1, @ref cpu_stm32f2, @ref cpu_stm32f3, @ref cpu_stm32f4,
  * @ref cpu_stm32l4, @ref cpu_stm32f7.
  *
- * STM32F[2|4|7] Clock configuration
+ * STM32F[0|1|2|4|7] Clock configuration
```
Comment (aabadie) [inline cpu/stm32_common/doc.txt]: f3 update missing here

File: cpu/stm32_common/doc.txt [hunk 3]
```diff
@@ -8,10 +8,11 @@
...
  * =================================
  *
- * stm32f2, stm32f4 and stm32f7 cpus share clock configuration code and macro.
+ * stm32f0, stm32f1, stm32f2, stm32f4 and stm32f7 cpus share clock configuration
```
Comment (aabadie) [inline cpu/stm32_common/doc.txt]: stm32f3 missing

File: cpu/stm32_common/doc.txt [hunk 4]
```diff
@@ -22,7 +23,12 @@
  *  - CLOCK_AHB_DIV, CLOCK_AHB: AHB prescaler in register value and AHB frequecny in Hz
  *  - CLOCK_APB1_DIV, CLOCK_APB1: APB1 prescaler in register value and APB1 frequecny in Hz
  *  - CLOCK_APB2_DIV, CLOCK_APB2: APB2 prescaler in register value and APB2 frequecny in Hz
+ * (CLOCK_APB2_DIV is not needed for stm32f0)
+ * The following macro must be defined for stm32f[2|4|7]:
  *  - CLOCK_PLL_M, CLOCK_PLL_N, CLOCK_PLL_P, CLOCK_PLL_Q: Main PLL factors
+ * The following macro must be defined for stm32f[0|1]:
```
Comment (aabadie) [inline cpu/stm32_common/doc.txt]: f3 missing

**Comment by aabadie** [comment]
Test results:
- [X] nucleo32-f042: works
- [X] nucleo-f072: works
- [X] nucleo-f103: works
- [X] nucleo32-f303: works
- [X] nucleo-f302: works
- [X] nucleo-f303: works
- [X] nucleo-f334: works
- [X] nucleo144-f303: works

**Comment by vincent-d** [comment]
Fixed doc, I also anticipated a bit #7537 in the doc

**Comment by vincent-d** [comment]
Thanks for the test effort @aabadie !

**Comment by aabadie** [comment]
Please squash

**Comment by vincent-d** [comment]
Squashed

**Comment by aabadie** [comment]
Murdock is green, go!

**Pull request merged**